Assets/Scripts/Map.cs

```diff
@@ -14,60 +25,157 @@ public class Map : MonoBehaviour
+            blocks[row, 0] = Instantiate(Resources.Load("Prefab/Blocks/" + blocksPath[((int)BlockType.Unbreakable)]) as GameObject).GetComponent<Block>();
+            blocks[row, sizeOfLevel - 1] = Instantiate(Resources.Load("Prefab/Blocks/" + blocksPath[((int)BlockType.Unbreakable)]) as GameObject).GetComponent<Block>();
+            tileMap.SetTile(new Vector3Int(0, -row, 0), blocks[row, 0].tile);
+            tileMap.SetTile(new Vector3Int(sizeOfLevel - 1, -row, 0), blocks[row, sizeOfLevel - 1].tile);
         }
+
+        generateChunck(new int[] { 1, 4, 8 });
     }
 
     // Update is called once per frame
     void Update()
     {
-        
+
     }
 
-    public int GetSize() {
+    public int GetSize()
+    {
         return sizeOfLevel;
     }
 
     public bool Break(int row, int col)
     {
-        var broken = blocks[row,col].onTap();
-        if(broken) {
-            blocks[row,col] = none;
-            tileMap.SetTile(new Vector3Int(col,-row,0),none.tile);
+        var broken = blocks[row, col].onTap();
+        if (broken)
+        {
+            blocks[row, col] = none;
+            tileMap.SetTile(new Vector3Int(col, -row, 0), none.tile);
         }
 
         return broken;
     }
+
+
+    public void generateChunck(int[] entryPoints)
+    {
+
+        int[][] chunk = new int[CHUNK_HEIGHT][];
+
+        for (var i = 0; i < CHUNK_HEIGHT; i++)
+        {
+            chunk[i] = new int[CHUNK_WIDTH];
+        }
+
+        for (var i = 0; i < 3; i++)
+        {
+            chunk[0][entryPoints[i]] = 1;
+            findPath(chunk, entryPoints[i]);
+        }
+
+        string matrix = "";
+
+        for (int i = 0; i < CHUNK_HEIGHT; i++)
+        {
+            for (int j = 0; j < CHUNK_WIDTH; j++)
+            {
+                matrix += (chunk[i][j] + "\t");
+            }
+            matrix += "\n";
+        }
+        Debug.Log(matrix);
+    }
+
+    void findPath(int[][] chunk, int xStart)
+    {
+        int row = 0;
+        int col = xStart;
+        Direction lastDirection = Direction.DOWN;
+
+        var downRate = 60.0;
+        var sideRate = (100.0 - downRate) / 2;
+
+        while (row != CHUNK_HEIGHT - 1)
+        {
+            int percentage = Random.Range(0, 100);
+            Direction currentDirection = Direction.DOWN;
+
+            if (percentage < downRate)
+            {
+                currentDirection = Direction.DOWN;
+            }
+            else if (percentage >= downRate && percentage < 100)
+            {
+                // I 
+                if (col == 0)
+                {
+                    currentDirection = lastDirection == Direction.DOWN ? Direction.RIGHT : Direction.DOWN;
+                }
+                else if (col == CHUNK_WIDTH - 1)
+                {
+                    currentDirection = lastDirection == Direction.DOWN ? Direction.LEFT : Direction.DOWN;
+                }
+
+                else if ((100 - percentage) < sideRate)
```

Comment:
Maybe this is an error. For example if we roll 90 with the random and we are on the edge, we enter in the else if line 142, set the currentDirection in line 147 or 151 and we overwrite the direction in line 156 because we enter in the else if line 154 too.